lib/reverse_sentence.rb

```diff
@@ -1,6 +1,52 @@
-# A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# A method to reverse the words in a sentence, in place
+
+# Time complexity: O(n) - my reverse_from_a_to_b method does not depend on the length of my_sentence,
+# the only method that is invoked and depends on the length of my_sentence is reverse_each_word_of_sentence
+
+# Space complexity: O(1) - the length of my_sentence does does not affect the space taken up to save our variable;
+# all strings take up the same space regardless of length; unlike arrays
+
 def reverse_sentence(my_sentence)
```

Comment:
👍 The method works and you have in-place space complexity.  I'm a little confused why you have `class << self` in there.  Otherwise well done.